pkg/gateway/model_build_targetgroup.go

```diff
@@ -240,36 +255,33 @@ func (t *targetGroupModelBuildTask) BuildTargetGroup(ctx context.Context) error
 			HealthCheckConfig:   healthCheckConfig,
 			IpAddressType:       ipAddressType,
 		},
-	}
-
-	tg := latticemodel.NewTargetGroup(t.stack, tgName, tgSpec)
-	t.log.Infof("buildTargetGroup, tg[%s], tgSpec %v", tgName, tg)
-
-	// add targetgroup to localcache for service reconcile to reference
-	// for serviceexport, the httproutename is set to ""
-	t.datastore.AddTargetGroup(tgName, "", "", "", tgSpec.Config.IsServiceImport, "")
+	})
+	t.log.Infof("buildTargetGroup, stackTG[%s], tgSpec%v \n", targetGroupName, stackTG)
+	t.datastore.AddTargetGroup(targetGroupName, "", "", "", false, "")
+	t.datastore.SetTargetGroupByServiceExport(targetGroupName, false, true)
+	return stackTG, nil
+}
 
-	if !t.serviceExport.DeletionTimestamp.IsZero() {
-		// triggered by serviceexport delete
-		t.datastore.SetTargetGroupByServiceExport(tgName, false, false)
-	} else {
-		// triggered by serviceexport add
-		t.datastore.SetTargetGroupByServiceExport(tgName, false, true)
+func (t *targetGroupModelBuildTask) buildTargetGroupForServiceExportDeletion(ctx context.Context, targetGroupName string) (*latticemodel.TargetGroup, error) {
+	stackTG := latticemodel.NewTargetGroup(t.stack, targetGroupName, latticemodel.TargetGroupSpec{
+		Name:      targetGroupName,
+		LatticeID: "",
+		IsDeleted: true,
+	})
+	t.datastore.SetTargetGroupByServiceExport(targetGroupName, false, false)
+	dsTG, err := t.datastore.GetTargetGroup(targetGroupName, "", false)
+	t.log.Infof("TargetGroup cached in datastore: %v \n", dsTG)
+	if err != nil {
+		return nil, fmt.Errorf("Cannot find targetgroup %v in Datastore,error :%v \n", targetGroupName, err)
```

Comment:
Please avoid use of "%v" on whatever logs and error messages, I know this is a refactoring PR but this is a new addition.

Reply:
Do you mean I should use `return fmt.Errorf("%w", err)` instead?

https://github.com/aws/aws-application-networking-k8s/blob/21643fd393896963c3a3cad9d41fd7f3fd7dd26e/pkg/gateway/model_build_lattice_service.go#L82